### PR TITLE
BYOND: Calling things each tick.

### PR DESCRIPTION
**Pull request by ghost**
Hello, i'm byond and i like to call Stat() each tick causing lag.

### CODE REVIEW

**Comment by Aranclanos** [comment]
yeah sure

**Comment by MrStonedOne** [comment]
:+1:

**Comment by Giacom** [comment]
So are all /mob/Stat()'s affected by this? I'm not sure what the stack flow of it is. Does client Stat() call the mob's Stat() or are they both called independently under the hood?

**Comment by Chnkr** [comment]
@Giacom No, there is one Stat for `/client/` and one for `/atom/` updated the pull to reflect the changes.

**Comment by Giacom** [comment]
Ah. You should then make sure all Stats() are calling ..() then, and make sure there is nothing time sensitive in there.

File: code/game/atoms.dm [hunk 1]
```diff
@@ -425,3 +425,11 @@ var/list/blood_splatter_icons = list()
...
 //This proc is called on the location of an atom when the atom is Destroy()'d
 /atom/proc/handle_atom_del(atom/A)
+
+// Byond seemingly calls stat, each tick.
+// Calling things each tick can get expensive real quick.
+// So we slow this down a little.
+// See: http://www.byond.com/docs/ref/info.html#/client/proc/Stat
+/atom/Stat()
+	. = ..()
+	sleep(1)
```
Comment (Giacom) [inline code/game/atoms.dm]: Maybe this could also be in a DEFINE somewhere. It might be worth tweaking it later.

Comment (Chnkr) [inline code/game/atoms.dm]: I wouldn't increase the sleep, doing that would result in things like the lobby timer, and other timers feeling real weird and janky.

**Comment by Razharas** [comment]
Is it good to go?
@Chnkr

**Comment by Chnkr** [comment]
@Razharas Yup! Shouldn't cause any issues.

**Pull request merged**